newlib/libc/sys/vita/fios2.h

```diff
@@ -0,0 +1,182 @@
+#ifndef __FIOS2_H__
+#define __FIOS2_H__
+
+#include <psp2/rtc.h>
+
+#define SCE_FIOS_OK 0
+
+#define SCE_FIOS_TIME_NULL ((SceFiosTime)0)
+#define SCE_FIOS_TIME_EARLIEST ((SceFiosTime)1)
+#define SCE_FIOS_TIME_LATEST ((SceFiosTime)0x7FFFFFFFFFFFFFFFLL)
+
+#define SCE_FIOS_PRIO_MIN ((int8_t)-128)
+#define SCE_FIOS_PRIO_DEFAULT ((int8_t)0)
+#define SCE_FIOS_PRIO_MAX ((int8_t)127)
+
+#define SCE_FIOS_FH_SIZE 80
+#define SCE_FIOS_DH_SIZE 80
+#define SCE_FIOS_OP_SIZE 168
+#define SCE_FIOS_CHUNK_SIZE 64
+
+#define SCE_FIOS_ALIGN_UP(val, align) (((val) + ((align) - 1)) & ~((align) - 1))
+#define SCE_FIOS_STORAGE_SIZE(num, size) (((num) * (size)) + SCE_FIOS_ALIGN_UP(SCE_FIOS_ALIGN_UP((num), 8) / 8, 8))
+
+#define SCE_FIOS_DH_STORAGE_SIZE(numDHs, pathMax) SCE_FIOS_STORAGE_SIZE(numDHs, SCE_FIOS_DH_SIZE + pathMax)
+#define SCE_FIOS_FH_STORAGE_SIZE(numFHs, pathMax) SCE_FIOS_STORAGE_SIZE(numFHs, SCE_FIOS_FH_SIZE + pathMax)
+#define SCE_FIOS_OP_STORAGE_SIZE(numOps, pathMax) SCE_FIOS_STORAGE_SIZE(numOps, SCE_FIOS_OP_SIZE + pathMax)
+#define SCE_FIOS_CHUNK_STORAGE_SIZE(numChunks) SCE_FIOS_STORAGE_SIZE(numChunks, SCE_FIOS_CHUNK_SIZE)
+
+#define SCE_FIOS_BUFFER_INITIALIZER  { 0, 0 }
+#define SCE_FIOS_PARAMS_INITIALIZER { 0, sizeof(SceFiosParams), 0, 0, 2, 1, 0, 0, 256 * 1024, 2, 0, 0, 0, 0, 0, SCE_FIOS_BUFFER_INITIALIZER, SCE_FIOS_BUFFER_INITIALIZER, SCE_FIOS_BUFFER_INITIALIZER, SCE_FIOS_BUFFER_INITIALIZER, NULL, NULL, NULL, { 66, 189, 66 }, { 0x40000, 0, 0x40000}, { 8 * 1024, 16 * 1024, 8 * 1024}}
+#define SCE_FIOS_OPENPARAMS_INITIALIZER { 0, 0, 0, SCE_FIOS_BUFFER_INITIALIZER }
+#define SCE_FIOS_OPATTR_INITIALIZER { 0, 0, 0, 0, 0, 0, 0, 0 }
+#define SCE_FIOS_RAM_CACHE_CONTEXT_INITIALIZER { sizeof(SceFiosRamCacheContext), 0, (64 * 1024), NULL, NULL, 0, {0, 0, 0} }
+
+typedef enum SceFiosWhence {
+  SCE_FIOS_SEEK_SET = 0,
+  SCE_FIOS_SEEK_CUR = 1,
+  SCE_FIOS_SEEK_END = 2
+} SceFiosWhence;
+
+typedef enum SceFiosOpenFlags {
+  SCE_FIOS_O_RDONLY = (1 << 0),
+  SCE_FIOS_O_WRONLY = (1 << 1),
+  SCE_FIOS_O_RDWR = (SCE_FIOS_O_RDONLY | SCE_FIOS_O_WRONLY),
+  SCE_FIOS_O_APPEND = (1 << 2),
+  SCE_FIOS_O_CREAT = (1 << 3),
+  SCE_FIOS_O_TRUNC = (1 << 4),
+} SceFiosOpenFlags;
+
+typedef int64_t SceFiosTime;
+
+typedef int32_t SceFiosFH;
+typedef int32_t SceFiosDH;
+typedef uint64_t SceFiosDate;
+typedef int64_t SceFiosOffset;
+typedef int64_t SceFiosSize;
+
+typedef struct SceFiosBuffer {
+  void *pPtr;
+  size_t length;
+} SceFiosBuffer;
+
+typedef struct SceFiosOpAttr {
+  SceFiosTime deadline;
+  void *pCallback;
+  void *pCallbackContext;
+  int32_t priority : 8;
+  uint32_t opflags : 24;
+  uint32_t userTag;
+  void *userPtr;
+  void *pReserved;
+} SceFiosOpAttr;
+
+typedef struct SceFiosRamCacheContext {
+  size_t sizeOfContext;
+  size_t workBufferSize;
+  size_t blockSize;
+  void *pWorkBuffer;
+  const char *pPath;
+  intptr_t flags;
+  intptr_t reserved[3];
+} SceFiosRamCacheContext;
```

Comment:
can we use `snake_case` instead of `camelCase`?
eg;

* `pPtr` -> `ptr`
* `pCallback` -> `callback`
* `pCallbackContext` -> `callback_context`
* `userTag` -> `user_tag` or just `tag`
* `userPtr` -> `user_ptr` or just `user`
* `pReserved` -> `reserved`
* `sizeOfContext` -> `size_of_context`
* `workBufferSize` -> `work_buffer_size`, `work_buf_size` or `buf_size`
* `blockSize` -> `block_size`
* `pWorkBuffer` -> `work_buffer`, `buffer` or just `buf`
* ...

Reply:
no problem. I'm still reversing some of the stuff and these are leftovers from vitashell.

Reply:
@TheOfficialFloW can you allow changing the license to LGPL or others?

Reply:
Yes